### PR TITLE
Refactor CI/CD

### PR DESCRIPTION
**Pull request by TessFerrandez**
- Move CI/CD into one section
- Restructure the folders and files so it's a bit more streamlined, moving recipes into recipes etc.
- Adding the why and the fundamentals
- Fixing links

### CODE REVIEW

File: docs/CI-CD/dev-sec-ops/secrets-management/README.md [hunk 1]
```diff
@@ -1,12 +1,24 @@
-# Secrets Management
+# Secrets management
 
-Secrets Management refers to the way in which we protect configuration settings and other sensitive data which, if
-made public, would allow unauthorized access to resources. Examples of secrets are usernames, passwords, api keys, SAS
-tokens etc.
+Secret Management refers to the tools and practices used to manage digital authentication credentials (like API keys, tokens, passwords, and certificates). These secrets are used to protect access to sensitive data and services, making their management critical for security.
```
Comment (Doommius) [inline docs/CI-CD/dev-sec-ops/secrets-management/README.md:3]: You made the header lower case, maybe this occurrence should be the same for consistency 
```suggestion
Secret management refers to the tools and practices used to manage digital authentication credentials (like API keys, tokens, passwords, and certificates). These secrets are used to protect access to sensitive data and services, making their management critical for security.
```

File: docs/CI-CD/recipes/reusing-devcontainers-within-a-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Options for building devcontainers within pipeline
+
+There are three ways to build devcontainers within pipeline:
+
+- With [GitHub - devcontainers/ci](https://github.com/devcontainers/ci) builds the container with the `devcontainer.json`. Example here: [devcontainers/ci · Getting Started](https://github.com/devcontainers/ci/blob/main/docs/github-action.md#getting-started).
+- With [GitHub - devcontainers/cli](https://github.com/devcontainers/cli), which is the same as the above, but using the underlying CLI directly without tasks.
+- Building the `DockerFile` with `docker build`. This option excludes all configuration/features specified within the `devcontainer.json`.
+
+## Considered Options
```
Comment (Doommius) [inline docs/CI-CD/recipes/reusing-devcontainers-within-a-pipeline.md:13]:
```suggestion
## Considered options
```

consistency, same as the above comment

**Review by Doommius — APPROVED** [review]
approved with suggestions

**Comment by TessFerrandez** [comment]
Merging with only 1 approver - as per conversation with @nyouens and EF team

**Pull request merged**